src/otx/v2/adapters/torch/mmengine/mmdet/engine.py

```diff
@@ -0,0 +1,214 @@
+"""OTX adapters.torch.mmengine.mmdet.Engine API."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import torch
+from mmdet.registry import VISUALIZERS
+
+from otx.v2.adapters.torch.mmengine.engine import MMXEngine
+from otx.v2.adapters.torch.mmengine.mmdet.registry import MMDetRegistry
+from otx.v2.adapters.torch.mmengine.modules.utils.config_utils import CustomConfig as Config
+from otx.v2.api.utils.logger import get_logger
+
+if TYPE_CHECKING:
+    import numpy as np
+
+logger = get_logger()
+
+
+class MMDetEngine(MMXEngine):
+    """The MMDetEngine class is responsible for running inference on pre-trained models."""
+
+    def __init__(
+        self,
+        work_dir: str | Path | None = None,
+    ) -> None:
```

Comment:
This seems to be less than 120 lines ?
```suggestion
    def __init__(self, work_dir: str | Path | None = None) -> None:
```